### PR TITLE
fix(admin-ui): add proper types

### PR DESCRIPTION
**Pull request by frdomovic**
# fix(admin-ui): add proper types

## Summary
Fix all @ts-ignore lines

## Summary by CodeRabbit

- **New Features**
	- Improved error handling for various scenarios during package and release addition.
	- Enhanced deploy status logic in the `PublishApplication` component.

- **Refactor**
	- Modified file reading and error logging in the `handleFileChange` function.
	- Adjusted the logic for setting deploy status based on response status.

- **Documentation**
	- Updated type declaration for `fileInputRef` in `useRef` to include `HTMLInputElement | null`.
	- Renamed `isFinalExecutionStatus` to `isFinalExecution` in `wallet.ts` with parameter and return type changes.

- **Chores**
	- Modified import statements in `wallet.ts` to include `FinalExecutionStatus` directly and added import for `nearAPI`.

### CODE REVIEW

File: node-ui/src/pages/PublishApplication.tsx [hunk 1]
```diff
@@ -107,13 +108,11 @@ export default function PublishApplication() {
   };
 
   const handleFileChange = (event: ChangeEvent<HTMLInputElement>) => {
-    // @ts-ignore: Object is possibly 'null'.
-    const file = event.target.files[0];
+    const file = event.target.files && event.target.files[0];
```
[the remark this answers is not in the document]
Comment (frdomovic) [reply node-ui/src/pages/PublishApplication.tsx:111]: updated

File: node-ui/src/pages/PublishApplication.tsx [hunk 2]
```diff
@@ -112,7 +112,7 @@ export default function PublishApplication() {
     if (file && file.name.endsWith(".wasm")) {
       const reader = new FileReader();
       reader.onload = async (e) => {
-        const arrayBuffer = new Uint8Array(e.target?.result as ArrayBufferLike);
+        const arrayBuffer = new Uint8Array(e?.target?.result as ArrayBufferLike);
```
Comment (MatejVukosav) [inline node-ui/src/pages/PublishApplication.tsx]: What if event is null? It doesn't look right just to ignore it

Comment (frdomovic) [reply node-ui/src/pages/PublishApplication.tsx]: Added console.error if its null, doesn't need popup status as this is just file change from input html component and recovery is just select again ...

Comment (MatejVukosav) [reply node-ui/src/pages/PublishApplication.tsx]: Can u early return in that case to avoid forward issues?

Comment (frdomovic) [reply node-ui/src/pages/PublishApplication.tsx]: updated

File: node-ui/src/pages/PublishApplication.tsx [hunk 3]
```diff
@@ -107,40 +108,48 @@ export default function PublishApplication() {
...
-    const file = event.target.files[0];
+    const file = event.target.files && event.target.files[0];
     if (file && file.name.endsWith(".wasm")) {
       const reader = new FileReader();
       reader.onload = async (e) => {
-        // @ts-ignore: Object is possibly 'null'.
-        const arrayBuffer = new Uint8Array(e.target.result as ArrayBufferLike);
-        const bytes = new Uint8Array(arrayBuffer);
-        const blob = new Blob([bytes], { type: "application/wasm" });
+        if (e && e.target && e.target.result) {
```
Comment (MatejVukosav) [inline node-ui/src/pages/PublishApplication.tsx]: it is the same as `e?.target?.result`

File: node-ui/src/pages/PublishApplication.tsx [hunk 4]
```diff
@@ -107,40 +108,48 @@ export default function PublishApplication() {
...
-          });
+          await axios
+            .post(BLOBBY_IPFS, blob)
+            .then((response) => {
+              setIpfsPath(`${BLOBBY_IPFS}/${response.data.cid}`);
+            })
+            .catch((error) => {
+              console.error("Error occurred while uploading the file:", error);
+            });
+        } else {
```
Comment (MatejVukosav) [inline node-ui/src/pages/PublishApplication.tsx]: remove the conditional scopes. Just make early check and keep code flat

Comment (MatejVukosav) [reply node-ui/src/pages/PublishApplication.tsx]: The same as you wrote algorithms or have params in function. If(something bad ) return .... then everything else

File: node-ui/src/pages/PublishApplication.tsx [hunk 5]
```diff
@@ -172,19 +181,25 @@ export default function PublishApplication() {
           },
         ],
       });
-      // @ts-expect-error: Property 'status' does not exist on type 'void | FinalExecutionOutcome'.
-      if (res.status.SuccessValue) {
+      if (
+        res &&
+        isFinalExecutionStatus(res.status) &&
+        res.status.SuccessValue
```
Comment (MatejVukosav) [inline node-ui/src/pages/PublishApplication.tsx]: Isn't this already verified in `isFinalExecutionStatus`? I would wrap all those checks into single function

File: node-ui/src/pages/PublishApplication.tsx (hunk 3 above)
Comment (frdomovic) [reply node-ui/src/pages/PublishApplication.tsx]: updated

File: node-ui/src/pages/PublishApplication.tsx (hunk 4 above)
Comment (frdomovic) [reply node-ui/src/pages/PublishApplication.tsx]: updated

File: node-ui/src/pages/PublishApplication.tsx (hunk 5 above)
Comment (frdomovic) [reply node-ui/src/pages/PublishApplication.tsx]: updated to single function

**Pull request merged**